passes/S038/testdata/src/a/main.go

```diff
@@ -0,0 +1,90 @@
+package a
+
+import "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+func f() {
+	// Comment ignored
+
+	//lintignore:S038
+	_ = &schema.Resource{
+		Read: func(*schema.ResourceData, interface{}) error { return nil },
+		Schema: map[string]*schema.Schema{
+			"x": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"foo": {
+							AtLeastOneOf: []string{"x.0.bar"},
+						},
+						"bar": {
+							AtLeastOneOf: []string{"x.0.foo"},
+						},
+					},
+				},
+			},
+		},
+	}
+	// Failing
```

Comment:
Let's ensure the following additional failure cases are tested:

- TypeSet
- Root attribute referencing non-existing root attribute
- Root attribute referencing existing configuration block attribute with non-existing nested attribute
- Configuration block nested attribute referencing non-existing root attribute
- Configuration block nested attribute referencing existing configuration block attribute with non-existing nested attribute

Reply:
Good point, and I will also add the corresponding pass test cases for those scenarios.